examples/parselog.rs

```diff
@@ -61,11 +61,7 @@ fn main() {
     loop {
         match handle.read(&mut buf) {
             Ok(0) => break,
-            Ok(n) => {
-                for byte in &buf[..n] {
-                    statemachine.advance(&mut performer, *byte);
-                }
-            },
+            Ok(n) => statemachine.advance(&mut performer, &buf[..n]),
```

Comment:
I would expect most people not to have trouble migrating, especially given how nice this example l's change is.

Reply:
> I would expect most people not to have trouble migrating, especially given how nice this example's change is.

Yes I agree that most people will likely have *less* code after migration, since most implementations already were using buffers anyway. That's one of the motivating factors behind this patch and also why I wanted to *intentionally* break things to make people aware of the faster API.